packages/firestore/src/core/bundle.ts

```diff
@@ -218,16 +218,17 @@ export class BundleLoader {
       'Bundled documents ends with a document metadata and missing document.'
     );
 
+    const result = await applyBundleDocuments(this.localStore, this.documents);
+
```

Comment:
As far as I can tell, returning the map of matching documents from `applyBundleDocuments` saves one iteration. This code would be less complex if you extracted the mapping in this function (ideally, via a call to a helper function). This would remove your changes from `applyBundleDocuments` and some of the changes from LocalStoreTester.

Reply:
Done. It IS nicer this way!